jina/executors/indexers/__init__.py

```diff
@@ -2,14 +2,17 @@
 __license__ = "Apache-2.0"
 
 import os
-from typing import Tuple, Union, List
+from typing import Tuple, Union, List, Iterator, Optional
 
 import numpy as np
 
 from .. import BaseExecutor
 from ..compound import CompoundExecutor
 from ...helper import call_obj_fn, cached_property, get_readable_size
 
+if False:
+    from ...proto import jina_pb2
+
 
 class BaseIndexer(BaseExecutor):
     """``BaseIndexer`` stores and searches with vectors.
```

Comment:
When you explicitly take out the vector docstring below, I'd also rewrite this to something like

"base class for storing and searching any kind of data strcuture"